pkg/registry/mock/registry.go

```diff
@@ -0,0 +1,51 @@
+package mock
```

Comment:
this is not entirely how mock should work. the class indeed should have a reference to `mock.Mock` but on each function, you explicitly tell the mock the function was called.

e.g.:


```golang
func (r *Registry) EnrichAndValidate() error {
	err := r.ValidateAbstractParams()
	if err != nil {
		return errors.Wrap(err, "Failed abstract registry params validation")
	}
	return nil
}
```

should be looking 

```golang
func (r *Registry) EnrichAndValidate() error {
	args := r.Called()
        return args.Error(0)
```

and when you want to assert it was called, from the mocked registry instance you will be doing

`registryInstance.On("EnrichAndValidate").Return(someExpectedErr).Once()`